### PR TITLE
More Improvements

### PR DESCRIPTION
**Pull request by darxriggs**
I suggest to have a look at the commits individually.

### CODE REVIEW

File: src/main/java/jenkinsci/plugins/influxdb/InfluxDbStep.java [hunk 1]
```diff
@@ -53,6 +53,7 @@ public String getSelectedTarget() {
         return target;
     }
 
+    @DataBoundSetter
```
Comment (asimell) [inline src/main/java/jenkinsci/plugins/influxdb/InfluxDbStep.java:56]: Are you sure this is necessary? According to https://jenkins.io/doc/developer/plugin-development/pipeline-integration/ only optional parameters need a `@DataboundSetter`:

> It is a good idea to replace a lengthy `@DataBoundConstructor` with a short one taking just truly mandatory parameters (such as a server location). For all optional parameters, create a public setter marked with `@DataBoundSetter` (with any non-null default value set in the constructor or field initializer).

File: pom.xml [hunk 2]
```diff
@@ -194,7 +194,7 @@ THE SOFTWARE.
         <dependency>
             <groupId>io.jenkins</groupId>
             <artifactId>configuration-as-code</artifactId>
-            <version>1.14</version>
+            <version>1.31</version>
```
Comment (asimell) [inline pom.xml:198]: Is there a reason for this dependency upgrade? Most of your other commits have a note about why some version was changed. Same comment applies for parent pom version.

Comment (darxriggs) [reply pom.xml:198]: Both upgrades are for staying up-to-date and to receive new features and bug fixes.
I had no specific changes in these dependencies in mind.

File: src/main/java/jenkinsci/plugins/influxdb/InfluxDbStep.java (hunk 1 above)
Comment (darxriggs) [reply src/main/java/jenkinsci/plugins/influxdb/InfluxDbStep.java:56]: I think in this case it's not mandatory as the parameter is already part of the constructor. But I think it doesn't hurt either. The goal was to make it consistent with a) the other setters and b) with `InfluxDbPublisher`.

**Comment by darxriggs** [comment]
I added 2 more commits.

**Comment by darxriggs** [comment]
@asimell is there anything I can support?

**Comment by asimell** [comment]
@darxriggs Not really, the changes look good. Unfortunately, I partly messed up my dev environment looking into https://issues.jenkins-ci.org/browse/JENKINS-59724, so I haven't been able to run all my test jobs regarding PRs. Sorry for the delay.

File: src/main/java/jenkinsci/plugins/influxdb/InfluxDbStep.java (hunk 1 above)
Comment (asimell) [reply src/main/java/jenkinsci/plugins/influxdb/InfluxDbStep.java:56]: Actually it's a bug to have it on both the setter and the constructor (ref. https://github.com/jenkinsci/kubernetes-plugin/pull/486), so I removed the `@DataboundSetter` from both InfluxDbStep and InfluxDbPublisher.

**Pull request merged**